### PR TITLE
Feature/nexus

### PR DESCRIPTION
**Pull request by Stayer**
Uploading artifacts (builds, bindings) to the Nexus server (https://nexus.iroha.tech/)

Linux

- [x] build
- [x] Java
- [x] Python2
- [ ] Python3 - blocked by https://soramitsu.atlassian.net/browse/DOPS-47
- [ ] Android - blocked by https://soramitsu.atlassian.net/browse/IR-1622

Windows - blocked by (ask @lebdron)
- [ ] Java
- [ ] Python

MacOS
- [x] build

### CODE REVIEW

File: .jenkinsci/artifacts.groovy [hunk 1]
```diff
@@ -24,32 +22,21 @@ def uploadArtifacts(filePaths, uploadPath, artifactServers=['artifact.soramitsu.
...
       // TODO @bakhtin 30.05.18 IR-1384. Make gpg command options and paths compatible with Windows OS.
       if (!agentType.contains('MSYS_NT')) {
-        sh "echo \"${CI_GPG_MASTERKEY}\" | $gpgKeyBinary -o \$(pwd)/\$(basename ${it}).asc ${it}"
-        sh "echo put \$(pwd)/\$(basename ${it}).asc $uploadPath >> \$(pwd)/batch.txt;"
+        sh "echo \"${CI_GPG_MASTERKEY}\" | $gpgKeyBinary -o \$(pwd)/\$(basename ${it}).ascfile ${it}"
+        sh "echo \$(pwd)/\$(basename ${it}).ascfile >> \$(pwd)/batch.txt;"
       }
-      sh "echo put \$(pwd)/\$(basename ${it}).sha256 $uploadPath >> \$(pwd)/batch.txt;"
-      sh "echo put \$(pwd)/\$(basename ${it}).md5 $uploadPath >> \$(pwd)/batch.txt;"
+      sh "echo \$(pwd)/\$(basename ${it}).sha256 >> \$(pwd)/batch.txt;"
```
Comment (bakhtin) [inline .jenkinsci/artifacts.groovy:33]: Nexus calculates these hashsums for you after file uploaded. So no need to upload it yourself

File: .jenkinsci/bindings.groovy [hunk 2]
```diff
@@ -29,30 +29,30 @@ def doJavaBindings(os, packageName, buildType=Release) {
...
   if (os == 'windows') {
-    sh "mkdir -p /tmp/${env.GIT_COMMIT}/bindings-artifact"
+    sh "mkdir -p /tmp/${GIT_COMMIT}/bindings-artifact"
     cmakeOptions = '-DCMAKE_TOOLCHAIN_FILE=/c/Users/Administrator/Downloads/vcpkg-master/vcpkg-master/scripts/buildsystems/vcpkg.cmake -G "NMake Makefiles"'
   }
   if (os == 'linux') {
     // do not use preinstalled libed25519
     sh "rm -rf /usr/local/include/ed25519*; unlink /usr/local/lib/libed25519.so; rm -f /usr/local/lib/libed25519.so.1.2.2"
   }
-  if (env.PBVersion == "python2") { supportPython2 = "ON" }
```
Comment (bakhtin) [inline .jenkinsci/bindings.groovy:55]: Why did you remove this line while it is still in use on line 61? `cmake` command would fail since variable is not defined.

File: Jenkinsfile [hunk 3]
```diff
@@ -9,14 +9,10 @@ properties([parameters([
   booleanParam(defaultValue: false, description: '', name: 'x86_64_win'),
   booleanParam(defaultValue: true, description: 'Build Java bindings', name: 'JavaBindings'),
   choice(choices: 'Release\nDebug', description: 'Java bindings build type', name: 'JBBuildType'),
-  string(defaultValue: 'jp.co.soramitsu.iroha', description: 'Java bindings package name', name: 'JBPackageName'),
-  booleanParam(defaultValue: true, description: 'Build Python bindings', name: 'PythonBindings'),
+  string(defaultValue: 'tech.iroha.libiroha', description: 'Java bindings package name', name: 'JBPackageName'),
```
Comment (bakhtin) [inline Jenkinsfile]: @nickaleks should we really change the namespace or leave it as it was?

File: Jenkinsfile [hunk 4]
```diff
@@ -73,124 +69,19 @@ pipeline {
             beforeAgent true
             expression { return params.x86_64_linux }
           }
-          agent { label 'x86_64' }
+          agent { label 'docker_1' }
```
Comment (bakhtin) [inline Jenkinsfile]: Don't forget to revert back to 'x86_64'

**Review by bakhtin — COMMENTED** [review]
Please, comment on which steps are required to finish with the rest of platforms/languages. You can also provide links on tasks if they block further progress. Thanks.

File: .jenkinsci/artifacts.groovy (hunk 1 above)
Comment (Stayer) [reply .jenkinsci/artifacts.groovy:33]: Yes, it does. But only in "browse" nexus mode. But in HTML View there are no hashes.
1. https://nexus.iroha.tech/#browse/browse:artifacts:libiroha%2Fbindings%2Fjava%2Fjava-bindings-Release-linux-20180902-fc5f88.zip
2. https://nexus.iroha.tech/service/rest/repository/browse/artifacts/libiroha/bindings/java/

File: .jenkinsci/bindings.groovy (hunk 2 above)
Comment (Stayer) [reply .jenkinsci/bindings.groovy:55]: I change it by this line:
https://github.com/hyperledger/libiroha/pull/5/files/57b1249e1293bc07f4a77e3b3155c53523568cf7#diff-13d8dd8a4496726a4cd57868d48d7040R43

`def supportPython2 = PBversion == "2" ? "ON" : "OFF"`

File: Jenkinsfile (hunk 3 above)
Comment (nickaleks) [reply Jenkinsfile]: Not sure about this. Probably we want to leave it as "jp.co.soramitsu.iroha"

**Review by nickaleks — APPROVED** [review]
Please resolve package name issue
